### PR TITLE
[MM-49782] Allow configurable deployment revision history limit

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

Allows for a configurable `Deployment.spec.revisionHistoryLimit` directly from the Mattermost spec.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-49782

#### Release Note

```release-note
Allow configurable deployment revision history limit
```

### CODE REVIEW

File: apis/mattermost/v1beta1/mattermost_types.go [hunk 1]
```diff
@@ -259,6 +263,13 @@ type PodTemplate struct {
 	ExtraLabels map[string]string `json:"extraLabels,omitempty"`
 }
 
+// PodTemplate defines configuration for the template for Mattermost pods.
```
Comment (mirshahriar) [inline apis/mattermost/v1beta1/mattermost_types.go:266]:
```suggestion
// DeploymentTemplate defines configuration for the template for Mattermost deployment.
```

File: apis/mattermost/v1beta1/mattermost_types.go [hunk 2]
```diff
@@ -120,6 +120,10 @@ type MattermostSpec struct {
 	// +optional
 	PodTemplate *PodTemplate `json:"podTemplate,omitempty"`
 
+	// DeploymentTemplate defines configuration for the template for Mattermost deployemnt.
```
Comment (mirshahriar) [inline apis/mattermost/v1beta1/mattermost_types.go:123]:
```suggestion
	// DeploymentTemplate defines configuration for the template for Mattermost deployment.
```

**Review by mirshahriar — APPROVED** [review]
Looks good. Just fixed minor typo

**Review by gabrieljackson — APPROVED** [review]
👍

**Pull request merged**